src/routes/docs/quick-starts/android-java/+page.markdoc

```diff
@@ -0,0 +1,393 @@
+{% /only_light %}
+
+If this is your first time using Appwrite, create an account and create your first project.
+
+Then, under **Add a platform**, add an **Android app**.
+
+Add your app's **name** and **package name**, your package name is the one entered when creating an Android project. For existing projects, you should use the **applicationId** in your app-level [build.gradle](https://github.com/appwrite/playground-for-android/blob/master/app/build.gradle#L11) file.
+
+{% only_dark %}
+![Add a platform](/images/docs/quick-starts/dark/add-platform.png)
+{% /only_dark %}
+{% only_light %}
+![Add a platform](/images/docs/quick-starts/add-platform.png)
+{% /only_light %}
+
+You can skip optional steps.
+
+{% /section %}
+
+{% section #step-3 step=3 title="Add the Appwrite SDK" %}
+To add the Appwrite SDK for Android as a dependency, add the following to your app-level **build.gradle** file inside the **dependencies** block.
+
+```java
+implementation "io.appwrite:sdk-for-android:7.0.0"
+```
+
+In order to allow creating OAuth sessions, the following activity needs to be added inside the `<application>` tag, along side the existing `<activity>` tags in your [AndroidManifest.xml](https://github.com/appwrite/playground-for-flutter/blob/master/android/app/src/main/AndroidManifest.xml).
+Be sure to replace the **[PROJECT_ID]** string with your actual Appwrite project ID.
+You can find your Appwrite project ID in you project settings screen in your Appwrite Console.
+
+```xml
+<manifest ...>
+  ...
+  <application ...>
+    ...
+    <!-- Add this inside the `<application>` tag, along side the existing `<activity>` tags -->
+    <activity android:name="io.appwrite.views.CallbackActivity" android:exported="true">
+      <intent-filter android:label="android_web_auth">
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data android:scheme="appwrite-callback-[PROJECT_ID]" />
+      </intent-filter>
+    </activity>
+  </application>
+</manifest>
+```
+{% /section %}
+
+{% section #step-4 step=4 title="Create Appwrite helper class" %}
+Find your project's ID in the **Settings** page.
+
+{% only_dark %}
+![Project settings screen](/images/docs/quick-starts/dark/project-id.png)
+{% /only_dark %}
+{% only_light %}
+![Project settings screen](/images/docs/quick-starts/project-id.png)
+{% /only_light %}
+
+Create a new file `AppwriteHelper.java` and add the following code to it, replacing `[PROJECT_ID]` with your project ID.
+
+```java
+package YOUR_ROOT_PACKAGE_HERE;
+
+import android.content.Context;
+
+import java.util.Map;
+
+import io.appwrite.Client;
+import io.appwrite.ID;
+import io.appwrite.coroutines.CoroutineCallback;
+import io.appwrite.models.Session;
+import io.appwrite.models.User;
+import io.appwrite.services.Account;
+
+public class AppwriteHelper {
+    private static Client client;
```

Comment:
I'd probably avoid using `static` vars and methods altogether.

I see we used `getApplicationContent()` later but `static` can still create some very weird and hard to debug issues.

We could instead make a singleton like this -
```java
private static AppwriteHelper instance;

public static synchronized AppwriteHelper getInstance(Context context) {
    if (instance == null) {
      instance = new AppwriteHelper(context.getApplicationContext());
    }

    return instance;
}
```

Reply:
Or maybe try using a `ViewModel` instead of a direct barebones class logic - https://developer.android.com/topic/libraries/architecture/viewmodel#java

Reply:
How's the update @ItzNotABug ?